spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/session/SessionAutoConfigurationJdbcTests.java

```diff
@@ -86,4 +88,15 @@ public void customTableName() {
 				.queryForList("select * from FOO_BAR")).isEmpty();
 	}
 
+	@Test
+	public void customTableNameWithDefaultSchemaFails() {
+		this.thrown.expect(BeanCreationException.class);
+		this.thrown.expectMessage(containsString(
+				"Database initializer cannot be configured"));
+		load(Arrays.asList(EmbeddedDataSourceConfiguration.class,
+				DataSourceTransactionManagerAutoConfiguration.class),
+				"spring.session.store-type=jdbc",
+				"spring.session.jdbc.table-name=FOO_BAR");
```

Comment:
> No, it's an exception when you configure custom table name together with default schema script location, those two together form an invalid configuration.

Correct me if I am wrong but you're not configuring the "default schema script location" so your sentence essentially can be rewritten as "it's an exception when you only configure a custom table name". That's also what your test shows. IMO that's wrong. Failing fast for something you haven't configured seems a bit harsh to me.

If you specify a custom table name, I wouldn't expect Spring Boot to auto-magically adapt the script (thought it could). We have other cases where we flip a switch based on other properties.

Maybe @wilkinsona has an opinion?